### PR TITLE
Fix layout_stride constructor for flexible stride array

### PR DESCRIPTION
**Pull request by crtrott**
Allow strides std::array to be any size type convertible.
I.e. follow other places where we take arrays.

### CODE REVIEW

File: P0009/P0009.md [hunk 1]
```diff
@@ -2000,61 +2002,68 @@ struct layout_stride {
...
+constexpr mapping(const Extents& e, array<SizeType, N> s) noexcept;
 ```
 [1]{.pnum} Let $P$ be a permutation of the integers $0, ...,$ `Extents::rank()-1` and let $p_i$ be the $i^{th}$ element of $P$.
 
-* [2]{.pnum} *Preconditions:*
+* [2]{.pnum} *Constraints:*
+
+    * [2.1]{.pnum} `is_convertible_v<SizeType, size_type>` is `true` and
+
+    * [2.2]{.pnum} `N == rank()` is `true`.
```
[the remark this answers is not in the document]
Comment (crtrott) [reply P0009/P0009.md:2014]: this is how mdspan works.

[the remark this answers is not in the document]
Comment (crtrott) [reply P0009/P0009.md:2014]: This is what the mdspan constructor does. The reason to do this is that you get an error message which says: your array doesn't have the right size, if I make it N you get: no overload found.

**Comment by crtrott** [comment]
Also fixed is_contiguous wording for layout_stride

File: P0009/P0009.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (mhoemmen) [reply P0009/P0009.md:2014]: Cool, that makes sense, thanks!

**Review by mhoemmen — APPROVED** [review]

**Pull request merged**